OrbitGl/FrameTrack.cpp

```diff
@@ -0,0 +1,228 @@
+      "%s times the average frame time for drawing purposes. These are drawn in red."
+      "<br/><br/>"
+      "<b>Note</b>: Timings are not the runtime of the function, but the difference "
+      "between start timestamps of subsequent calls."
+      "<br/><br/>"
+      "<b>Frame marker function:</b> %s<br/>"
+      "<b>Module:</b> %s<br/>"
+      "<b>Maximum frame time:</b> %s<br/>"
+      "<b>Minimum frame time:</b> %s<br/>"
+      "<b>Average frame time:</b> %s<br/>",
+      function_name, std::to_string(kHeightCapAverageMultipleUint64), function_name,
+      FunctionUtils::GetLoadedModuleName(function_),
+      GetPrettyTime(absl::Nanoseconds(stats_.max_ns())),
+      GetPrettyTime(absl::Nanoseconds(stats_.min_ns())),
+      GetPrettyTime(absl::Nanoseconds(stats_.average_time_ns())));
+}
+
+std::string FrameTrack::GetBoxTooltip(PickingId id) const {
+  const TextBox* text_box = time_graph_->GetBatcher().GetTextBox(id);
+  if (!text_box) {
+    return "";
+  }
+  // TODO(wotzlaw): Support case for manual instrumentation.
+  std::string function_name = FunctionUtils::GetDisplayName(function_);
+
+  return absl::StrFormat(
+      "<b>Frame time</b><br/>"
+      "<i>Frame time based on two subsequent calls to %s. Height and width of the box are "
+      "proportional to time where height is capped at %s times the average time. Timeslices with "
+      "capped height are shown in red.</i>"
+      "<br/><br/>"
+      "<b>Frame marker function:</b> %s<br/>"
+      "<b>Module:</b> %s<br/>"
+      "<b>Frame time:</b> %s",
+      function_name, std::to_string(kHeightCapAverageMultipleUint64), function_name,
+      FunctionUtils::GetLoadedModuleName(function_),
+      GetPrettyTime(
+          TicksToDuration(text_box->GetTimerInfo().start(), text_box->GetTimerInfo().end())));
+}
+
+void FrameTrack::Draw(GlCanvas* canvas, PickingMode picking_mode) {
+  TimerTrack::Draw(canvas, picking_mode);
+
+  const Color kWhiteColor(255, 255, 255, 255);
+  const Color kBlackColor(0, 0, 0, 255);
+
+  Batcher* batcher = &time_graph_->GetBatcher();
+  float y = pos_[1] - GetMaximumBoxHeight() + GetAverageBoxHeight();
+  float x = pos_[0];
+  Vec2 from(x, y);
+  Vec2 to(x + size_[0], y);
+  float ui_z = GlCanvas::kZValueUi;
+
+  const TimeGraphLayout& layout = time_graph_->GetLayout();
+  std::string avg_time = GetPrettyTime(absl::Nanoseconds(stats_.average_time_ns()));
+  std::string label = absl::StrFormat("Avg: %s", avg_time);
+
+  int string_height_unused = 0;
+  int string_width = 0;
+  canvas->GetTextRenderer().GetStringSize(label.c_str(), string_width, string_height_unused);
+
+  // It seems the size of the box that contains the text needs to be a little larger than the
+  // text size itself. Not adding the 1.f on the line results in the last character of the string
+  // not getting rendered.
+  Vec2 white_text_box_size(static_cast<float>(string_width) + 1.f, layout.GetTextBoxHeight());
```

Comment:
I prefer to add this in GetStringSize. Let me add that, as my PR is almost mergeable.

Reply:
Fixed to use GetStringWidth().